shell/common/rasterizer.cc

```diff
@@ -421,14 +421,11 @@ RasterStatus Rasterizer::DrawToSurface(flutter::LayerTree& layer_tree) {
       return raster_status;
     }
     if (external_view_embedder != nullptr) {
+      FML_DCHECK(!frame->IsSubmitted());
       external_view_embedder->SubmitFrame(surface_->GetContext(),
-                                          root_surface_canvas);
-      // The external view embedder may mutate the root surface canvas while
-      // submitting the frame.
-      // Therefore, submit the final frame after asking the external view
-      // embedder to submit the frame.
-      frame->Submit();
-      external_view_embedder->FinishFrame();
+                                          std::move(frame));
+      // The frame must be submitted at this point.
+      FML_DCHECK(frame->IsSubmitted());
```

Comment:
This is a use after move. This check is passing because of undefined behavior.

Reply:
Done.